traits/observers/_trait_added_observer.py

```diff
@@ -0,0 +1,258 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+from traits.observers._trait_change_event import trait_event_factory
+from traits.observers._has_traits_helpers import (
+    iter_objects,
+    object_has_named_trait,
+)
+from traits.observers._i_observer import IObserver
+from traits.observers._observe import add_or_remove_notifiers
+from traits.observers._observer_change_notifier import ObserverChangeNotifier
+from traits.observers._observer_graph import ObserverGraph
+
+
+@IObserver.register
+class TraitAddedObserver:
+    """ An observer for observing the trait_added event.
+
+    This observer only offers the "maintainer". When this observer is used in
+    an ObserverGraph, its subgraphs are the graphs to be hooked up when a new
+    trait is added, provided that the trait satisfies a given criterion.
+    The criterion should align with the root observer(s) in these subgraph(s).
+
+    Parameters
+    ----------
+    match_func : callable(str, TraitType) -> boolean
```

Comment:
`TraitType` doesn't seem like the right thing here. (We haven't got rid of `TraitHandler` yet, so not all of our traits even have an associated `TraitType`.)

I think I'd expect to receive the actual trait (so an instance of `CTrait`) instead.

Reply:
Thanks!  This is a mistake indeed.  Fixed.